avm-transpiler/src/transpile_contract.rs

```diff
@@ -0,0 +1,120 @@
+use log::info;
+use regex::Regex;
+use serde::{Deserialize, Serialize};
+
+use acvm::acir::circuit::Circuit;
+use noirc_driver::ContractFunctionType;
+
+use crate::transpile::brillig_to_avm;
+use crate::utils::extract_brillig_from_acir;
+
+/// Representation of a contract with some transpiled functions
+#[derive(Debug, Serialize, Deserialize)]
+pub struct TranspiledContract {
+    pub transpiled: bool,
+    pub noir_version: String,
+    pub name: String,
+    // Functions can be ACIR or AVM
+    pub functions: Vec<AvmOrAcirContractFunction>,
+    pub events: serde_json::Value,
+    pub file_map: serde_json::Value,
+    //pub warnings: serde_json::Value,
+}
+
+/// A regular contract with ACIR+Brillig functions
+/// but with fields irrelevant to transpilation
+/// represented as catch-all serde Values
+#[derive(Debug, Serialize, Deserialize)]
+pub struct CompiledAcirContract {
+    pub noir_version: String,
+    pub name: String,
+    pub functions: Vec<AcirContractFunction>,
+    pub events: serde_json::Value,
+    pub file_map: serde_json::Value,
+    //pub warnings: serde_json::Value,
+}
+
+/// Transpilation is performed when a TranspiledContract
+/// is constructed from a CompiledAcirContract
+impl From<CompiledAcirContract> for TranspiledContract {
```

Comment:
functionally this is good I would perfer to have all of the types on top

Reply:
What exactly do you mean by that?

Reply:
Just layout, the method is sandwiched by the types, rather than the types being together